pennylane/wires.py

```diff
@@ -427,11 +448,26 @@ def unique_wires(list_of_wires):
                     "Expected a Wires object; got {} of type {}.".format(wires, type(wires))
                 )
 
+        label_sets = [wire.toset() for wire in list_of_wires]
+        seen_ever = set()
+        seen_once = set()
+
+        # Find unique set in O(n) time.
+        for labels in label_sets:
+            # (seen_once ^ labels) finds all of the unique labels seen once
+            # (seen_ever - seen_once) is the set of labels already seen more than once
+            # Subtracting these two sets makes a set of labels only seen once so far.
+            seen_once = (seen_once ^ labels) - (seen_ever - seen_once)
+            # Update seen labels with all new seen labels
+            seen_ever.update(labels)
+
+        # Get unique values in order they appear.
         unique = []
         for wires in list_of_wires:
-            for wire in wires:
+            for wire in wires.tolist():
                 # check that wire is only contained in one of the Wires objects
-                if sum([1 for wires_ in list_of_wires if wire in wires_]) == 1:
+                if wire in seen_once:
                     unique.append(wire)
+                    seen_once.remove(wire)
 
         return Wires(unique)
```

Comment:
This last update I had some confusion about :thinking:

Reply:
Actually we don't need this line since we already know it was only seen once.